### PR TITLE
Fix error: mmcsd/mmcsd_sdio.c:2669:35: error: 'buffer' may be used uninitialized [-Werror=maybe-uninitialized]

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since new version gcc toolchain make more check.

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: drivers/mmcsd/mmcsd_sdio.c [hunk 1]
```diff
@@ -2659,6 +2659,8 @@ static int mmcsd_read_csd(FAR struct mmcsd_state_s *priv)
       return -EPERM;
     }
 
+  memset(buffer, sizeof(buffer));
```
Comment (pkarashchenko) [inline drivers/mmcsd/mmcsd_sdio.c:2662]:
```suggestion
  memset(buffer, 0, sizeof(buffer));
```

Comment (xiaoxiang781216) [reply drivers/mmcsd/mmcsd_sdio.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Still some other errors remain

**Comment by xiaoxiang781216** [comment]
> Still some other errors remain

Fix here: https://github.com/apache/incubator-nuttx/pull/7398/commits/8b78bcc9649d26b27a82394e7f6a19c68a386553
The new gcc version do the great job to catch so many error in compile time.

**Comment by pkarashchenko** [comment]
> > Still some other errors remain
> 
> Fix here: https://github.com/apache/incubator-nuttx/pull/7398/commits/8b78bcc9649d26b27a82394e7f6a19c68a386553
> The new gcc version do a great job to catch so many error in compile time.

Yes. But I was expecting builds to be run locally and those fixes to be merged prior to toolchain upgrade :)

**Comment by xiaoxiang781216** [comment]
It's hard to compile all config in the local machine

**Review by pkarashchenko — APPROVED** [review]

**Comment by btashton** [comment]
> It's hard to compile all config in the local machine. It's more simple that ci use the new docker image(instead the old) run all config if the patch change Dockerfile. @btashton is it possible?

Unfortunately you cannot push from a PR to the container registry (which makes sense for security), but what you can do it push to a registry including one off your GitHub fork, and then reference that in the PR until things are working. I have done that a few times. 

I can also put some more thought into other workflow setups but this is probably the easiest path for now.

File: arch/arm/src/sam34/sam4s_nand.c [hunk 2]
```diff
@@ -149,7 +149,11 @@ static int nand_wait_ready(struct sam_nandcs_s *priv)
 
   /* The ready/busy (R/nB) signal of the NAND Flash  */
 
-  while (!sam_gpioread(priv->rb));
+  while (!sam_gpioread(priv->rb))
+    {
+      ;
```
Comment (pkarashchenko) [inline arch/arm/src/sam34/sam4s_nand.c:154]:
```suggestion
```

Comment (xiaoxiang781216) [reply arch/arm/src/sam34/sam4s_nand.c]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 seems like the new toolchain has an issue with `gcov`:
```
rm-none-eabi-ld: /github/workspace/sources/nuttx/staging/libarch.a(rp2040_start.o): in function `__start':
/github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_start.c:83: undefined reference to `__gcov_indirect_call_profiler_v4'
arm-none-eabi-ld: /github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_start.c:139: undefined reference to `__gcov_indirect_call'
arm-none-eabi-ld: /github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_start.c:139: undefined reference to `__gcov_time_profiler_counter'
arm-none-eabi-ld: /github/workspace/sources/nuttx/staging/libarch.a(rp2040_clock.o): in function `rp2040_clock_configure':
/github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_clock.c:87: undefined reference to `__gcov_indirect_call_profiler_v4'
arm-none-eabi-ld: /github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_clock.c:101: undefined reference to `__gcov_topn_values_profiler'
arm-none-eabi-ld: /github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_clock.c:144: undefined reference to `__gcov_topn_values_profiler'
arm-none-eabi-ld: /github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_clock.c:147: undefined reference to `__gcov_indirect_call'
arm-none-eabi-ld: /github/workspace/sources/nuttx/arch/arm/src/chip/rp2040_clock.c:147: undefined reference to `__gcov_time_profiler_counter'
...
```
Please investigate. Most probably `EXTRA_LIBS += $(wildcard $(shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libgcov.a))` is not evaluated properly.

Anyway I'm merging this RP because `gcov` issue seems to be the last step to restore CI on mainline

**Comment by xiaoxiang781216** [comment]
@anchao is looking this problem, look like the toolchain compile libgcov with the wrong config. Anyway, if we can't fix the problem soon, we can temp disable gcov option from defconfig.

**Pull request merged**